File: jim/fizzbuzz.rb
#
# ruby stream processing approach to FizzBuzz problem.
#
puts (1..100) \
  .map {|x| [x,'']} \
  .map {|pair| pair[0] % 3 == 0 ? [pair[0], 'Fizz'] : pair} \
  .map {|pair| pair[0] % 5 == 0 ? [pair[0], pair[1] + 'Buzz'] : pair} \
  .map {|pair| pair[1] == '' ? pair[0] : pair[1]} \
  .join("\n")

File: everest/FizzBuzz.java
public class FizzBuzz {

	public static void main(String[] args) {
		
		for (int number = 1; number <= 100; number++ ){
			if (number % 3 == 0 && number % 5 == 0){
				System.out.println("FizzBuzz");
			}
			else if (number % 3 == 0){
				System.out.println("Fizz");
			}
			else if (number % 5 == 0){
				System.out.print("Buzz");
			}
			else {
				System.out.println(number);
			}
		}

	}

}